pkg/userop/middleware.go

```diff
@@ -82,102 +81,52 @@ func getGasPricesMiddleware(provider EthBackend, gasConfig GasConfig) middleware
 
 		logger.Debug("getting gas prices")
 
-		// Calculate maxPriorityFeePerGas
 		var maxPriorityFeePerGas *big.Int
-		if !op.MaxPriorityFeePerGas.IsZero() {
-			maxPriorityFeePerGas = op.MaxPriorityFeePerGas.BigInt()
-		} else {
-			var maxPriorityFeePerGasStr string
-			if err := provider.RPC().CallContext(ctx, &maxPriorityFeePerGasStr, "eth_maxPriorityFeePerGas"); err != nil {
-				return err
-			}
-
-			var ok bool
-			maxPriorityFeePerGas, ok = new(big.Int).SetString(maxPriorityFeePerGasStr, 0)
-			if !ok {
-				return fmt.Errorf("failed to parse maxPriorityFeePerGas: %s", maxPriorityFeePerGasStr)
-			}
-			logger.Debug("fetched maxPriorityFeePerGas", "maxPriorityFeePerGas", maxPriorityFeePerGas.String())
+		var maxFeePerGas *big.Int
 
-			// Increase maxPriorityFeePerGas to give user more
-			// flexibility in setting the gas price.
-			maxPriorityFeePerGas.Mul(
-				maxPriorityFeePerGas,
-				gasConfig.MaxPriorityFeePerGasMultiplier.BigInt())
+		chainId, err := provider.ChainID(ctx)
+		if err != nil {
+			return fmt.Errorf("failed to get chain ID: %w", err)
 		}
 
-		// Calculate maxFeePerGas
-		var maxFeePerGas *big.Int
-		if !op.MaxFeePerGas.IsZero() {
-			maxFeePerGas = op.MaxFeePerGas.BigInt()
-		} else {
-			// Get the latest block to read its base fee
-			block, err := provider.BlockByNumber(ctx, nil)
+		isPolygon := chainId.Cmp(big.NewInt(137)) == 0 || chainId.Cmp(big.NewInt(80002)) == 0
```

Comment:
There's a way to simplify integer comparisons by casting `chainId` to `uint64` with `chainId.Uint64()`

Reply:
Thanks for the suggestion 👍

Reply:
68cb8e64bff038323cf947fc2e017df1c8dafce5